### ISSUE TITLE
Language Adapter Training Queries

### ISSUE THREAD

**Issue by pugantsov** [adapter-hub/adapter-transformers#457]
I was looking to train a language adapter using HuggingFace's AutoModelForMaskedLM (BERT) and I was wondering how this differs from a Task Adapter setup. I can see that in the docs, there is a subsection entitled _Train a Language Adapter_ but I noticed the link to the guide was broken. I had a look at the Github repo under the `pytorch/language-modeling` subsection of the `examples` folder and I was a little confused as I don't use the HuggingFace Trainer (I have my own Pytorch-based setup in my script) and a lot of the inputs are based on arguments supplied by the user so I was hoping for possibly a little more information from here.

Am I right in assuming that the setup for training a language adapter is virtually identical to that of say, a classification adapter? For example, something as straightforward as:

```
model.add_adapter(
    MODEL_NAME, config=transformers.adapters.PfeifferConfig())
model.train_adapter(MODEL_NAME)
model.set_active_adapters(MODEL_NAME)
```

I was also wondering about the differences in configs that might be worth considering when running a masked language modelling task specifically. For instance, I see on the docs that between the task and language adapter argument setup, the config is switched to `pfeiffer+inv` as opposed to `pfeiffer`. Is this simply down to preference or is this `pfeiffer+inv` config better for CLM and MLM tasks?

Apologies if I've missed something glaring obvious in the guide itself and this has been answered there.

**Comment by hSterz** [comment]
Yes, adding and training language adapters works as you described. If you want to use a language adapter in a multilingual setup it is beneficial to use an invertible adapter (e.g. `pfeiffer+inv`). The intuition behind that is that the invertible adapter should mitigate the mismatch between the vocabulary of the source and the target language (for more detail and an illustration [this](https://arxiv.org/pdf/2005.00052.pdf) is the paper where it was introduced).  If you want to use an invertible adapter, they are trained together with the bottleneck adapter on MLM or CLM. Hence it is sufficient to add the `pfeiffer+inv` adapter and train it as you described. I hope this helps.

**Comment by pugantsov** [comment]
Perfect, thank you so much!
